src/Microsoft.AspNetCore.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs

```diff
@@ -151,27 +152,33 @@ internal static void AddRazorViewEngineServices(IServiceCollection services)
             //
             // Razor compilation infrastructure
             //
-            services.TryAddSingleton<RazorProject, FileProviderRazorProject>();
-            services.TryAddSingleton<RazorTemplateEngine, MvcRazorTemplateEngine>();
             services.TryAddSingleton<LazyMetadataReferenceFeature>();
-
+            services.TryAddSingleton<RazorProjectFileSystem, FileProviderRazorProjectFileSystem>();
             services.TryAddSingleton(s =>
             {
-                return RazorEngine.Create(b =>
+                var fileSystem = s.GetRequiredService<RazorProjectFileSystem>();
+                var projectEngine = RazorProjectEngine.Create(RazorConfiguration.Default, fileSystem, builder =>
                 {
-                    RazorExtensions.Register(b);
+                    RazorExtensions.Register(builder);
 
                     // Roslyn + TagHelpers infrastructure
                     var metadataReferenceFeature = s.GetRequiredService<LazyMetadataReferenceFeature>();
-                    b.Features.Add(metadataReferenceFeature);
-                    b.Features.Add(new Microsoft.CodeAnalysis.Razor.CompilationTagHelperFeature());
+                    builder.Features.Add(metadataReferenceFeature);
+                    builder.Features.Add(new CodeAnalysis.Razor.CompilationTagHelperFeature());
```

Comment:
Is there a reason we don't use a `using` here?

Reply:
Just copy and pasted; will change.